### PR TITLE
Routing, CRUD function example, dynamic user detail example, simple search

### PR DESCRIPTION
**Pull request by fnaufalh**
I create some feature examples to get visibility on our system. This is the list:

- [x] Routing
  - [x] Single Route
  - [x] Nested Route
  - [x] Dynamic Route
- [x] CRUD
  - [x] Change Roles
  - [x] Update User Details
- [x] Simple Search (with case sensitive and start from first letter/word, cannot search from mid or end letter/word )

### CODE REVIEW

File: src/repositories/SearchRepository.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  public async search(keyword: any) {
+    const userQuery = query(
+      collection(database, 'user'),
+      where('name', '>=', keyword),
+      where('name', '<=', keyword + '\uf8ff')
+    )
+    const roleQuery = query(
+      collection(database, 'role'),
+      where('name', '>=', keyword),
+      where('name', '<=', keyword + '\uf8ff')
```
Comment (yusakwijaya) [inline src/repositories/SearchRepository.ts:14]: symbol greater than equal dan symbol less than equal itu fungsinya buat apa ya?

File: src/router.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const routes = [
+  {
+    path: '/',
+    component: Home
+  },
+  {
+    path: '/user',
+    component: User,
+    name: 'User'
```
Comment (yusakwijaya) [inline src/router.ts:17]: di router ini ada yg pake name: value, trus ada jg yg ngga, name ini berarti buat apa ya?

File: src/repositories/SearchRepository.ts (hunk 1 above)
Comment (fnaufalh) [reply src/repositories/SearchRepository.ts:14]: it's an alternative way to use 'like' methods
`>= name is have another character more than keyword
<= name can be less than keyword with maximum range of unicode

you can try it on firebase using menu query builder

File: src/router.ts (hunk 2 above)
Comment (fnaufalh) [reply src/router.ts:17]: an alternative way to call the path
e.g. <router-link :to="{name: User ...}" or <router-link :to="/user" for a simple way

mostly name use when we need complex routing or only to give identifier about the path

**Review by yusakwijaya — APPROVED** [review]

**Pull request merged**